openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -368,3 +375,93 @@ def test_xblock_handlers(self, xblock_type, xblock_name, user, status_code):
         self.client.login(username=self.users[user].username, password=TEST_PASSWORD)
         response = self.client.post(url)
         self.assertEqual(response.status_code, status_code)
+
+
+@ddt.ddt
+@override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+@override_settings(FIELD_OVERRIDE_PROVIDERS=(
+    'openedx.features.content_type_gating.field_override.ContentTypeGatingFieldOverride',
+))
+class TestConditionalContentAccess(TestConditionalContent):
+    @classmethod
+    def setUpClass(cls):
+        super(TestConditionalContentAccess, cls).setUpClass()
+        cls.factory = RequestFactory()
+    
+    def setUp(self):
+        super(TestConditionalContentAccess, self).setUp()
+
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='audit')
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='verified')
+
+        # Audit users have already been created. Now create verified users, and partition.
+        self.student_verified_a = UserFactory.create(username='student_verified_a', email='student_verified_a@example.com')
+        CourseEnrollmentFactory.create(user=self.student_verified_a, course_id=self.course.id, mode='verified')
+        self.student_verified_b = UserFactory.create(username='student_verified_b', email='student_verified_b@example.com')
+        CourseEnrollmentFactory.create(user=self.student_verified_b, course_id=self.course.id, mode='verified')
+
+        # Use this to put users into content gating groups
+        UserCourseTagFactory(
+            user=self.student_verified_a,
+            course_id=self.course.id,
+            key='xblock.partition_service.partition_{0}'.format(self.partition.id),
+            value=str(self.user_partition_group_a)
+        )
+        UserCourseTagFactory(
+            user=self.student_verified_b,
+            course_id=self.course.id,
+            key='xblock.partition_service.partition_{0}'.format(self.partition.id),
+            value=str(self.user_partition_group_b)
+        )
+
+        self.block_a = ItemFactory.create(
+            parent=self.vertical_a,
+            category='problem',
+            display_name='problem_a',
+            graded=True,
+            weight=1,
+        )
+        self.block_b = ItemFactory.create(
+            parent=self.vertical_b,
+            category='problem',
+            display_name='problem_b',
+            graded=True,
+            weight=1,
+        )
+
+
+    @ddt.unpack
```

Comment:
i don't think this is currently doing anything (though you will need to use this and the @ddt.ddt above if you do add ddt parameters to this function).  
I would be happy to walk through an example of how we use ddt if that helps

Reply:
yep. true.